types/types.go

```diff
@@ -2,9 +2,50 @@ package types
 
 import (
 	"encoding/json"
+	"fmt"
 	"strconv"
 )
 
+// Uint64 is a wrapper for uint64, but it is marshalled to and from JSON as a string
+type Uint64 uint64
+
+func (u Uint64) MarshalJSON() ([]byte, error) {
+	return json.Marshal(strconv.FormatUint(uint64(u), 10))
+}
+
+func (u *Uint64) UnmarshalJSON(data []byte) error {
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return fmt.Errorf("cannot unmarshal %s into Uint64, expected string-encoded integer", data)
+	}
+	v, err := strconv.ParseUint(s, 10, 64)
+	if err != nil {
+		return fmt.Errorf("cannot unmarshal %s into Uint64, expected string-encoded integer", data)
```

Comment:
Could you use different error messages for the two error cases? In case of debugging it usually helps to know at which step the operation failed, especially since nobody really uses debuggers and we don't have stack traces.